R/dataframe.R

```diff
@@ -157,6 +160,9 @@ as_data_frame.data.frame <- function(x, ...) {
 #' @export
 #' @rdname as_data_frame
 as_data_frame.list <- function(x, validate = TRUE, ...) {
+
+  x <- compact(x)
```

Comment:
Not meaning to criticize dplyr -- rather pointing out that this is one of those corner cases...

If I leave out this line, the call `data_frame(a = NULL)` already throws the error "Each variable must be a 1d atomic vector or list.". No need for further action (13b0e49).

data.frame(a = NULL) swallows a, but data.frame(a = NULL, b = 1:3) throws an error. I think there's no need to be consistent with data.frame() here.